src/test/kotlin/uk/gov/justice/hmpps/offenderevents/services/HMPPSDomainEventsEmitterTest.kt

```diff
@@ -1243,99 +1246,158 @@ internal class HMPPSDomainEventsEmitterTest(@Autowired private val objectMapper:
-      assertThatJson(payload).node("description")
-        .isEqualTo("A prisoner visitor restriction record has changed")
+    @DisplayName("will raise prison-offender-events.visitor.restriction.deleted event")
+    @Test
+    fun willRaiseDeletedDomainEvent() {
+      with(deletedPayload) {
+        assertJsonPath("eventType", "prison-offender-events.visitor.restriction.deleted")
+        assertJsonPath("occurredAt", "2022-12-04T10:00:00Z")
+        assertJsonPathDateTimeIsCloseTo("publishedAt", java.time.OffsetDateTime.now(), within(10, java.time.temporal.ChronoUnit.SECONDS))
+        assertJsonPath("personReference.identifiers[0].type", "PERSON")
+        assertJsonPath("personReference.identifiers[0].value", "4")
+        assertJsonPath("additionalInformation.visitorRestrictionId", "1")
+        assertJsonPath("additionalInformation.restrictionType", "DIHCON")
+        assertJsonPath("additionalInformation.effectiveDate", "2022-12-04")
+        assertJsonPath("additionalInformation.expiryDate", "2022-12-05")
+        assertJsonPath("additionalInformation.personId", "4")
+      }
     }
 
-    @Test
-    fun `will add correct fields to telemetry event`() {
-      assertThat(telemetryAttributes).containsEntry("occurredAt", "2022-12-04T10:00:00Z")
-      assertThat(telemetryAttributes).containsEntry("restrictionType", "SEC")
+    @DisplayName("will raise prison-offender-events.visitor.restriction.changed event")
+    @Test
+    fun willRaiseChangeDomainEvent() {
+      with(changedPayload) {
+        assertJsonPath("eventType", "prison-offender-events.visitor.restriction.changed")
+        assertJsonPath("occurredAt", "2022-12-04T10:00:00Z")
+        assertJsonPathDateTimeIsCloseTo("publishedAt", java.time.OffsetDateTime.now(), within(10, java.time.temporal.ChronoUnit.SECONDS))
+        assertJsonPath("personReference.identifiers[0].type", "PERSON")
+        assertJsonPath("personReference.identifiers[0].value", "4")
+        assertJsonPath("additionalInformation.visitorRestrictionId", "1")
+        assertJsonPath("additionalInformation.restrictionType", "DIHCON")
+        assertJsonPath("additionalInformation.effectiveDate", "2022-12-04")
+        assertJsonPath("additionalInformation.expiryDate", "2022-12-05")
+        assertJsonPath("additionalInformation.personId", "4")
+      }
     }
 
     @Test
-    fun `will contain no other telemetry properties`() {
-      assertThat(telemetryAttributes).containsOnlyKeys(
-        "eventType",
-        "occurredAt",
-        "publishedAt",
-        "personId",
-        "restrictionType",
-        "effectiveDate",
-        "expiryDate",
-        "visitorRestrictionId",
-        "enteredById",
-      )
+    fun `will raise telemetry events`() {
+      assertThat(deletedTelemetry).containsEntry("eventType", "prison-offender-events.visitor.restriction.deleted")
+      assertThat(changedTelemetry).containsEntry("eventType", "prison-offender-events.visitor.restriction.changed")
```

Comment:
Looks like the old telemetry assertions were a bit more complicated - not worth it?